### DISCUSSION TITLE
`test_dependency_gets_exception`, test failure in debian unstable

### DISCUSSION THREAD

**Discussion by sandrotosi** [fastapi/fastapi#9934] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
$ python3.11 -m pytest tests/test_dependency_normal_exceptions.py::test_dependency_gets_exception
```

### Description

When running the test suite during fastapi debian package build, the test `test_dependency_gets_exception` fails with:

```
    def test_dependency_gets_exception():
        assert state["except"] is False
        assert state["finally"] is False
        response = client.put("/invalid-user/rick", json="Morty")
        assert response.status_code == 400, response.text
        assert response.json() == {"detail": "Invalid user"}
>       assert state["except"] is True
E       assert False is True
```

and indeed at the end `state == {'except': False, 'finally': True}`

from my limited knowledge and quick debugging, it appears as if https://github.com/tiangolo/fastapi/blob/master/tests/test_dependency_normal_exceptions.py#L21-L22 is never called. I tried to capture also `Exception` but still nothing.

### Operating System

Linux

### FastAPI Version

0.100.0

### Python Version

Python 3.11.4

**Comment by Kludex** [comment, marked as answer]
The latest Starlette is being used on your environment. It shouldn't... 

See https://github.com/tiangolo/fastapi/pull/9636 for more details.

**Reply by sandrotosi** [reply]
thanks for the pointer to #9636 

the Debian packaging system allows for only a single version to be available, which in the case of starlette currently is 0.30.0; for fastapi i've currently disabled that test, as all the others are passing and so that i can upload 0.100.0 to Debian
